src/main/java/net/minecraftforge/common/crafting/GrindStoneRecipe.java

```diff
@@ -0,0 +1,110 @@
+package net.minecraftforge.common.crafting;
+
+import com.google.gson.JsonObject;
+
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.item.crafting.ShapedRecipe;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.JSONUtils;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+
+public class GrindStoneRecipe implements IRecipe<IInventory>
+{
+
+	private ResourceLocation id;
+	private Ingredient ingredient;
+	private Ingredient ingredient1;
+	private ItemStack result;
+	
+
+	public GrindStoneRecipe(ResourceLocation id,Ingredient ingredient, Ingredient ingredient1, ItemStack result) 
+	{
+		this.id = id;
+		this.ingredient = ingredient;
+		this.ingredient1 = ingredient1;
```

Comment:
What is ingredient1?

Reply:
The grindstone has 2 input slots, so their are 2 ingredients. I will probably change it to a list, similar to Shapeless, unless someone thinks its better to keep them separate? Than I will find some better names.

Reply:
Yes, but those are not the name of the slots you gave it, you called it `base` and `addition`, which does not match the naming scheme here. Plus, they're just two generic ingredients as the grindstone could be considered a shapeless recipe with only two slots. It doesn't matter which goes in which. So, actually you'll probably need to reimplement the checking logic as well.

Reply:
Yes I agree, like I said, I copied the smithing table for it, which uses those tags. I will change it to match what it is for the grindstone.